libp2p/Host.cpp

```diff
@@ -694,8 +706,15 @@ void Host::run(boost::system::error_code const&)
 	}
 
 	for (auto p: toConnect)
-		if (p->peerType == PeerType::Required && reqConn++ < m_idealPeerCount)
+	{
+		bool peerTypeIsRequired{};
+		{
+			RecursiveGuard l(x_sessions);
```

Comment:
this looks weird... `toConnect` is local list of shared_ptrs, do you try to protect Peer with a mutex for sessions?

Maybe let's try to make `Node::peerType` atomic, will it help?

Reply:
Yes, that worked.